model/postgres/postgres_server.rb

```diff
@@ -74,14 +87,36 @@ def configure_hash
         }
       },
       identity: resource.identity,
-      hosts: "#{vm.ephemeral_net4} #{resource.identity}"
+      hosts: "#{resource.representative_server.vm.ephemeral_net4} #{resource.identity}"
     }
   end
 
   def primary?
     timeline_access == "push"
   end
 
+  def standby?
+    timeline_access == "fetch" && !doing_pitr?
+  end
+
+  def doing_pitr?
+    !resource.representative_server.primary?
+  end
+
+  def failover_target
+    targets = resource.servers.select(&:standby?)
+    targets.reject! { _1.strand.label != "wait" }
+    targets.map! { [_1, _1.vm.sshable.cmd("sudo -u postgres psql -At -c 'SELECT pg_last_wal_receive_lsn()'").chomp] }
+    target = targets.max_by { lsn2int(_1[1]) }
+
+    if resource.ha_type == PostgresResource::HaType::ASYNC
+      return nil if lsn_monitor.last_known_lsn.nil?
+      return nil if lsn_diff(lsn_monitor.last_known_lsn, target[1]) > 80 * 1024 * 1024 # 80 MB or ~5 WAL files
+    end
+
+    target[0]
```

Comment:
nit: Why do we save it to a variable and doing in place operations instead of chaining?
Also I prefer hash instead of array.


```suggestion
    target = resource.servers
        .select{ _1.standby? && _1.strand.label == "wait" }
        .map { {server: _1, lsn: _1.vm.sshable.cmd("sudo -u postgres psql -At -c 'SELECT pg_last_wal_receive_lsn()'").chomp} }
        .max_by { lsn2int(_1[:lsn]) }

    if resource.ha_type == PostgresResource::HaType::ASYNC
      return nil if lsn_monitor.last_known_lsn.nil?
      return nil if lsn_diff(lsn_monitor.last_known_lsn, target[:lsn]) > 80 * 1024 * 1024 # 80 MB or ~5 WAL files
    end

    target[:server]
```